velox/connectors/Connector.h

```diff
@@ -421,6 +421,13 @@ class ConnectorFactory {
 /// FB_ANONYMOUS_VARIABLE.
 bool registerConnectorFactory(std::shared_ptr<ConnectorFactory> factory);
 
+/// Adds an alias for an existing factory. Throws if factory with the
+/// connectorName is not already present. Throws if the alias for the
+/// connectorName is already present.
+bool registerConnectorFactoryAlias(
```

Comment:
I'm wondering if it would be better to change ConnectorFactory API to allow for aliases. Perhaps, add 

```
const std::vector<std::string>& aliases() const;
```

method.

Or, modify registerConnectorFactory API to add an optional 'aliases' parameter. That would be similar to aliases allowed when registering scalar functions.

Reply:
|  I'm wondering if it would be better to change ConnectorFactory API to allow for aliases.

@mbasmanova This was my initial implementation, but then we register the Velox connectors in the Velox library. 
So an application such as Prestissimo would have to manually change the Velox code to include additional aliases. For example, to add "iceberg", we have to manually modify the alias list.
I then added this API so that we can add aliases in Prestissimo without having to change Velox code.

Reply:
See the commit for your approach https://github.com/facebookincubator/velox/commit/a30aa7f9d0b03dc232faa1b4cf787df4980fc26a

Reply:
It sounds like Velox auto-registers connector factories and this is problematic. Let's change that to require applications to explicitly register what they need.